innovation.game.php

```diff
@@ -11164,6 +11195,29 @@ function stInteractionStep() {
             );
             break;
 
+        // id 157, Artifacts age 5: Bill of Rights
+        case "157C1A":
+            $more_vis_cards = array();
+            for($color = 0; $color < 5; $color++){
+                if (self::countVisibleCards($player_id, $color) > self::countVisibleCards($launcher_id, $color))
+                {
+                    $more_vis_cards[] = $color;
+                }
+            }
+        
+            $options = array(
+                'player_id' => $player_id,
+                'n' => 1,
+                'can_pass' => false,
+                'color' => $more_vis_cards,
+                
+                'owner_from' => $player_id,
+                'location_from' => 'board',
+                'owner_to' => $launcher_id,
+                'location_to' => 'none'
```

Comment:
We can simplify the implementation this card by using `'choose_color' => true`. For example, see "Mask of Warka".

Reply:
If you follow "Mask of Warka" closely, you can figure out where the other code for this card belongs.

Reply:
Never mind, the suggestion won't work, since that doesn't support choosing a subset of colors, and it looks non-trivial to extend support for that.